### PR TITLE
added docker-compose file

### PR DESCRIPTION
**Pull request by FarisHijazi**
simple docker file for development. It has a volume to use the code and also a volume for ~/.cache/huggingface so that it can use downloaded models

### CODE REVIEW

**Review by ehartford — COMMENTED** [review]
I don't see the purpose of a docker-compose when we only have/need a single container

**Review by winglian — APPROVED** [review]
lgtm

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    environment:
+      - WANDB_API_KEY=${WANDB_API_KEY}
+    deploy:
+      resources:
+        reservations:
+          devices:
+            - driver: nvidia
+              # count: 1
+              capabilities: [gpu]
+    command: tail -f /dev/null
```
Comment (winglian) [inline docker-compose.yaml:20]: is this the correct command for local dev?

Comment (FarisHijazi) [reply docker-compose.yaml:20]: `command: tail -f /dev/null` keeps the container running otherwise it will shutdown immediately

**Comment by FarisHijazi** [comment]
> I don't see the purpose of a docker-compose when we only have/need a single container

if you have a way to just use a docker build/run command I'm more than happy to use that

File: docker-compose.yaml (hunk 1 above)
Comment (winglian) [reply docker-compose.yaml:20]: 👍 sgtm, I may change it to use sleep infinity down the line, https://stackoverflow.com/a/35770783

**Comment by winglian** [comment]
Thanks @FarisHijazi !!!

**Pull request merged**